### PR TITLE
Add zero length check on indices during NextSyncCommitteeIndices

### LINKED ISSUE #13051: panic: runtime error: integer divide by zero - then database corrupted - hangs at "Checking DB" forever taking 200+ GB Memory
### Describe the bug

[I got this error then the database is corrupted and never recovers. ](panic: runtime error: integer divide by zero - then database corrupted - hangs at "Checking DB" forever taking 200+ GB Memory)
```
Oct 13 16:12:22 hostname bash[406365]: panic: runtime error: integer divide by zero
Oct 13 16:12:22 hostname bash[406365]: goroutine 859 [running]:
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/consensus-types/primitives.ValidatorIndex.Mod(...)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/consensus-types/primitives/validator.go:39
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/altair.NextSyncCommitteeIndices({0x25fdd38, 0xc020eeb740}, {0x2628f28?, 0xc0d1632fc0})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/altair/sync_committee.go:129 +0x4d8
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/altair.NextSyncCommittee({0x25fdd38?, 0xc020eeb740?}, {0x2628f28?, 0xc0d1632fc0})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/altair/sync_committee.go:64 +0x50
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/altair.ProcessSyncCommitteeUpdates({0x25fdd38, 0xc020eeb740}, {0x2628f28?, 0xc0d1632fc0})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/altair/epoch_spec.go:33 +0xcd
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/altair.ProcessEpoch({0x25fdd38?, 0xc066752ea0?}, {0x2628f28, 0xc0d1632fc0})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/altair/transition.go:108 +0x3a9
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/transition.ProcessSlots({0x25fdd38?, 0xc066752e70?}, {0x2628f28, 0xc06728cfc0}, 0x72e986)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/transition/transition.go:258 +0xa93
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/core/transition.ProcessSlotsUsingNextSlotCache({0x25fdd38?, 0xc066752cf0?}, {0x2628f28, 0xc06728cfc0}, {0xc0f21c88a0, 0x20, 0x20}, 0x72e986)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/core/transition/transition.go:159 +0x185
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain.(*Service).getPayloadAttribute(0xc00024f770, {0x25fdd38, 0xc066752cf0}, {0x2628f28, 0xc0c2566d80}, 0x72e986, {0xc0f21c88a0, 0x20, 0x20})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/blockchain/execution_engine.go:271 +0x17a
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain.(*Service).notifyForkchoiceUpdate(0xc00024f770, {0x25fdd38?, 0xc0400fe4b0?}, 0xc12da2a890)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/blockchain/execution_engine.go:73 +0x725
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain.(*Service).onBlockBatch(0xc00024f770, {0x25fdd38?, 0xc0400fe480?}, {0xc00345b800, 0x40, 0x0?}, {0xc0c1d9d800, 0x40, 0x0?})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/blockchain/process_block.go:329 +0x1b52
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain.(*Service).ReceiveBlockBatch(0xc00024f770, {0x25fdc90?, 0xc005f4dbd0?}, {0xc00345b800?, 0x40, 0x40}, {0xc0c1d9d800, 0x40, 0x40})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/blockchain/receive_block.go:182 +0x1db
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/sync/initial-sync.(*Service).processBatchedBlocks(0xc0000f3590, {0x25fdc90, 0xc005f4dbd0}, {0x0?, 0x0?, 0x3a800e0?}, {0xc00345b800, 0x40, 0x40}, 0xc0f6163328)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/sync/initial-sync/round_robin.go:288 +0x8c4
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/sync/initial-sync.(*Service).processFetchedData(0xc00024f770?, {0x25fdc90?, 0xc005f4dbd0?}, {0x0?, 0xed7582757?, 0x3a800e0?}, 0x427e3d?, 0xc093a26030?)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/sync/initial-sync/round_robin.go:131 +0x11e
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/sync/initial-sync.(*Service).syncToFinalizedEpoch(0xc0000f3590, {0x25fdc90, 0xc005f4dbd0}, {0x1ca35c0?, 0x37fcb20?, 0x3a800e0?})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/sync/initial-sync/round_robin.go:84 +0x285
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/sync/initial-sync.(*Service).roundRobinSync(0xc0000f3590, {0xe00000004?, 0xc00681ba20?, 0x3a800e0?})
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/sync/initial-sync/round_robin.go:47 +0x19b
Oct 13 16:12:22 hostname bash[406365]: github.com/prysmaticlabs/prysm/v4/beacon-chain/sync/initial-sync.(*Service).Start(0xc0000f3590)
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/beacon-chain/sync/initial-sync/service.go:118 +0x8ff
Oct 13 16:12:22 hostname bash[406365]: created by github.com/prysmaticlabs/prysm/v4/runtime.(*ServiceRegistry).StartAll
Oct 13 16:12:22 hostname bash[406365]:         /home/prysm/prysm/runtime/service_registry.go:46 +0xec
Oct 13 16:12:23 hostname bash[405995]: exit status 2
Oct 13 16:12:23 hostname systemd[1]: prysm-mainnet-beacon.service: Main process exited, code=exited, status=1/FAILURE
Oct 13 16:12:23 hostname systemd[1]: prysm-mainnet-beacon.service: Failed with result 'exit-code'.
Oct 13 16:12:23 hostname systemd[1]: prysm-mainnet-beacon.service: Consumed 2w 1d 13h 36min 56.756s CPU time.
Oct 13 16:12:28 hostname systemd[1]: prysm-mainnet-beacon.service: Scheduled restart job, restart counter is at 31.
Oct 13 16:12:28 hostname systemd[1]: Stopped prysm-mainnet-beacon.
Oct 13 16:12:28 hostname systemd[1]: prysm-mainnet-beacon.service: Consumed 2w 1d 13h 36min 56.756s CPU time.
Oct 13 16:12:28 hostname systemd[1]: Started prysm-mainnet-beacon.
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=info msg="Finished reading JWT secret from /home/prysm/prysm-mainnet-beacon/jwt.hex"
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Running on Ethereum Mainnet" prefix=flags
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" prefix=flags slasher="Enables a slasher in the beacon node for detecting slashable offenses"
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" enable-historical-state-representation="Enables the beacon chain to save historical states in a space efficient manner. (Warning): Once enabled, this feature migrates your database in to a new schema and there is no going back. At worst, your entire database might get corrupted." prefix=flags
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" prefix=flags save-full-execution-payloads="Saves beacon blocks with full execution payloads instead of execution payload headers in the database"
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" enable-verbose-sig-verification="Enables identifying invalid signatures if batch verification fails when processing block" prefix=flags
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" enable-optional-engine-methods="Enables the optional engine methods" prefix=flags
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Enabled feature flag" prefix=flags prepare-all-payloads="Informs the engine to prepare all local payloads. Useful for relayers and builders"
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Subscribing to All Attestation Subnets" prefix=flags
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=warning msg="Setting 128 slots per archive point and 4096 max RPC page size for historical slasher usage. This requires additional storage" prefix=node
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=info msg="Default fee recipient is set to 0xa8fF27bcF3b0d54325c8DC658071626A6f1E9090, recipient may be overwritten from validator client and persist in db. Default fee recipient will be used as a fall back" prefix=node
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=info msg="Checking DB" database-path="/home/prysm/prysm-mainnet-beacon/beaconchaindata" prefix=node
Oct 13 16:12:50 hostname bash[2507892]: time="2023-10-13 16:12:50" level=info msg="Opening Bolt DB at /home/prysm/prysm-mainnet-beacon/beaconchaindata/beaconchain.db" prefix=db
Oct 13 16:12:52 hostname bash[2507892]: time="2023-10-13 16:12:52" level=warning msg="Saving full beacon blocks to the database. For greater disk space savings, we recommend resyncing from an empty database with checkpoint sync to save only blinded beacon blocks by default" prefix=db
Oct 13 16:12:53 hostname bash[2507892]: time="2023-10-13 16:12:53" level=info msg="Deposit contract: 0x00000000219ab540356cbb839cbe05303d7705fa" prefix=node
Oct 13 16:12:53 hostname bash[2507892]: time="2023-10-13 16:12:53" level=info msg="Checking DB" database-path="/home/prysm/prysm-mainnet-beacon/beaconchaindata" prefix=node
```

### Has this worked before in a previous version?

_No response_

### 🔬 Minimal Reproduction

Command:
```
CGO_CFLAGS="-O -D__BLST_PORTABLE__" CGO_CFLAGS_ALLOW="-O -D__BLST_PORTABLE__"\
    go run ./cmd/beacon-chain\
  --slots-per-archive-point=1  \
  --enable-historical-state-representation \
  --enable-optional-engine-methods \
  --subscribe-all-subnets \
  --slasher-datadir=$(eval echo ~$USER)/beacon-node \
  --accept-terms-of-use \
  --save-full-execution-payloads \
  --prepare-all-payloads \
  --historical-slasher-node \
  --slasher \
  --enable-verbose-sig-verification \
  --datadir=$(eval echo ~$USER)/beacon-node \
  --rpc-host=0.0.0.0 \
  --grpc-gateway-host=0.0.0.0 \
  --monitoring-host=0.0.0.0 \
  --jwt-secret=$(eval echo ~$USER)/jwt.hex \
  --suggested-fee-recipient=0xa8fF27bcF3b0d54325c8DC658071626A6f1E9090 \
  --execution-endpoint=http://localhost:9101
```

### Error

_No response_

### Platform(s)

Linux (x86)

### What version of Prysm are you running? (Which release)

_No response_

### Anything else relevant (validator index / public key)?

_No response_

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

Prevents a divide by zero panic

**Which issues(s) does this PR fix?**

Fixes #13051

**Other notes for review**

### CODE REVIEW

**Review by potuz — DISMISSED** [review]
LGTM

**Comment by prestonvanloon** [comment]
Some tests may be failing due to improper test setup. I'll fix shortly

File: beacon-chain/rpc/prysm/v1alpha1/beacon/assignments_test.go [hunk 1]
```diff
@@ -45,44 +44,6 @@ func TestServer_ListAssignments_CannotRequestFutureEpoch(t *testing.T) {
 	assert.ErrorContains(t, wanted, err)
 }
 
-func TestServer_ListAssignments_NoResults(t *testing.T) {
```
Comment (prestonvanloon) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/assignments_test.go:48]: This is no longer possible. This scenario was only possible when a beacon state had zero validators. All other scenarios are handled with API errors such as validator index out of bounds, validator public key not found, etc.

File: beacon-chain/rpc/prysm/v1alpha1/validator/aggregator_test.go [hunk 2]
```diff
@@ -74,8 +74,8 @@ func TestSubmitAggregateAndProof_IsAggregatorAndNoAtts(t *testing.T) {
 	s, err := state_native.InitializeFromProtoPhase0(&ethpb.BeaconState{
 		RandaoMixes: make([][]byte, params.BeaconConfig().EpochsPerHistoricalVector),
 		Validators: []*ethpb.Validator{
-			{PublicKey: pubKey(0)},
-			{PublicKey: pubKey(1)},
+			{PublicKey: pubKey(0), ExitEpoch: params.BeaconConfig().FarFutureEpoch},
+			{PublicKey: pubKey(1), ExitEpoch: params.BeaconConfig().FarFutureEpoch},
```
Comment (prestonvanloon) [inline beacon-chain/rpc/prysm/v1alpha1/validator/aggregator_test.go:78]: A future exit epoch was required for the validators to be considered active.

**Review by potuz — APPROVED** [review]

**Pull request merged**